src/ensureApiEnabled.ts

```diff
@@ -40,6 +40,12 @@ export async function check(
   return isEnabled;
 }
 
+const apiPermissionDeniedRegex = new RegExp(/Permission denied to enable service \[([.a-zA-Z]+)\]/);
```

Comment:
This is not relevant for the CLI (since the RegExp is freshly loaded on each command), but may be for VSCode - repeatedly calling `exec` on the same RegExp can lead to some unexpected results, since exec sets RegExp.lastIndex (https://stackoverflow.com/questions/11477415/why-does-javascripts-regex-exec-not-always-return-the-same-value)

If the VSCode extension might hit this error multiple times from a long running process, consider instantiating this RegExp inside of `enable`

Reply:
Thanks for the catch, I've always used `match` but eslint seems to prefer `exec` so this is my first time using it, good to learn about the differences.